### PR TITLE
Source Freshdesk: handle rate limiting #1972

### LINKED ISSUE #1972: Source Freshdesk: handle rate limiting
## Tell us about the problem you're trying to solve
If the connector hits any of the rate limits it should back off and retry after sleep

### PR DESCRIPTION
**Pull request by eugene-kulak**
## What
#1972

## How
*Describe the solution*

## Pre-merge Checklist
- [ ] *Run integration tests*
- [ ] *Publish Docker images*

## Recommended reading order
1. `test.java`
1. `component.ts`
1. the rest

Note: Based on #2052

### CODE REVIEW

**Comment by eugene-kulak** [comment]
This backoff strategy works similarly to other connectors.
Freshdesk API returns the number of seconds that needs to the client when he reaches the limit.
I don't know how to use it with backoff library yet...
@sherifnada

**Review by sherifnada — COMMENTED** [review]
@eugene-kulak imo it's ok if we don't use the backoff library. Can we use a custom decorator?

File: airbyte-integrations/bases/base-python/base_python/source.py [hunk 1]
```diff
@@ -69,28 +74,36 @@ def check(self, logger: AirbyteLogger, config: json) -> AirbyteConnectionStatus:
...
 
     def read(
-        self, logger: AirbyteLogger, config: json, catalog: ConfiguredAirbyteCatalog, state: Dict[str, any]
+        self, logger: AirbyteLogger, config: Mapping[str, Any], catalog: ConfiguredAirbyteCatalog, state: Dict[str, any]
     ) -> Generator[AirbyteMessage, None, None]:
         client = self._get_client(config)
 
-        logger.info(f"Starting syncing {self.__class__.__name__}")
+        logger.info(f"Starting syncing {self.name}")
         total_state = {**state}
```
[the remark this answers is not in the document]
Comment (eugene-kulak) [reply airbyte-integrations/bases/base-python/base_python/source.py:84]: good catch, yes

**Review by sherifnada — APPROVED** [review]

File: airbyte-integrations/connectors/source-freshdesk/source_freshdesk/utils.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        on_backoff=log_retry_attempt,
+        giveup=giveup_handler,
+        **kwargs,
+    )
+
+
+def retry_after_handler(**kwargs):
+    """Retry helper when we hit the call limit, sleeps for specific duration"""
+
+    def log_retry_attempt(_details):
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-freshdesk/source_freshdesk/utils.py:58]: this should be called `sleep_on_ratelimit`

**Pull request merged**